modules/javafx.web/src/test/java/test/javafx/scene/web/EventListenerLeakTest.java

```diff
@@ -627,13 +649,370 @@ public void TestStrongRefNewContentLoad() throws Exception {
+        Thread.sleep(100);
+        assertEquals("Click count", 3, listeners.get(0).get().getClickCount());
+
+        domNodes1.clear();
+        webView1 = null;
+
+        submit(() -> {
+            // Send clilck events to webview2
+            click(webView2, 0);
+            click(webView2, 1);
+            click(webView2, 2);
+        });
+
+        // Verify that the events are delivered to webview2
+        Thread.sleep(100);
+        assertEquals("Click count", 4, listeners.get(0).get().getClickCount());
+        assertEquals("Click count", 4, listeners.get(2).get().getClickCount());
+        // Verify that listener is still strongly held
+        assertNumActive("listeners", listenerRefs, 3);
+
+        //removed shared listener
+        submit(() -> {
+            domNodes2 = getDomNodes(webView2);
+            domNodes2.get(2).removeEventListener("click", listeners.get(4).get(), false);
+        });
+
+        submit(() -> {
+            // Send clilck events to webview2
+            click(webView2, 2);
+        });
+
+        // Verify that the event is not delivered to webview2 for shared listener
+        Thread.sleep(100);
+        assertEquals("Click count", 2, listeners.get(4).get().getClickCount());
+
+        // check active listeners
+        listeners.clear();
+        domNodes2.clear();
+        webView2 = null;
+        Thread.sleep(100);
+        // Verify that active listener
+        assertNumActive("MyListener", listenerRefs, 0);
+        listenerRefs.clear();
+    }
+
+    /**
+     * Test that multiple listeners on same node
+     */
+    @Test
+    public void oneWebViewMultipleListenerSameNode() throws Exception {
+        webView2 = null; // unused
+
+        // Load HTML content and get list of DOM nodes
+        loadContent(webView1, HTML);
+
+        final List<MyListener> listeners = new ArrayList<>();
+        submit(() -> {
+            domNodes1 = getDomNodes(webView1);
+            assertEquals(NUM_DOM_NODES, domNodes1.size());
+
+            // Create listener and attach to DOM node 0
+            MyListener myListener1 = MyListener.create();
+            MyListener myListener2 = MyListener.create();
+            listeners.add(myListener1);
+            listeners.add(myListener2);
+            domNodes1.get(0).addEventListener("click", listeners.get(0), false);
+            domNodes1.get(0).addEventListener("click", listeners.get(1), false);
+
+            // Send clilck event
+            click(webView1, 0);
+        });
+
+        Thread.sleep(100);
+        // Verify that the event is delivered to the listener
+        assertEquals("Click count", 1, listeners.get(0).getClickCount());
+        assertEquals("Click count", 1, listeners.get(0).getClickCount());
+
+        submit(() -> {
+            // Remove event listener
+            domNodes1.get(0).removeEventListener("click", listeners.get(0), false);
+            // Send clilck event
+            click(webView1, 0);
+        });
+
```

Comment:
It would be useful to call `assertNumActive' and verify that there is one active ref.